### PR TITLE
Add support for Eurofighter and Rafale mod aircraft

### PR DESCRIPTION
**Pull request by angelatthetomb**
Edited db.py, faction.py, start_generator.py, ai_flight_generator.py, qmodsettings.py, qnewgamegenerator.py. Added eurofighter.py and rafale.py. Added squadrons and custom loadouts. Tested in my forked build and work in-game.

### CODE REVIEW

**Review by Khopa — APPROVED** [review]

**Comment by Khopa** [comment]
Hello, 

Thanks, the code changes looks good :)

However, i'm unsure which mod version is being used, can you confirm if it's these one : 

- Rafale : https://forums.eagle.ru/topic/252774-rafales-pack-new-version-005-for-279-by-the-cuesta-brothers/
- EuroFighter : https://forums.eagle.ru/topic/290086-new-version-eurofighter-typhoon-by-lechuzas-negras-v105/

**Comment by Khopa** [comment]
💹 I've been able to test it succesfully with the Typhoon mod.

Just two things : 
* UI icons are missing
* There are no default factions allowing the use of the mod.

image

Could you add some for the modded planes ? (in resources/ui/aircrafts/icons)
Should be named {aircraft_class_id}_24.jpg and be a 91x24 jpeg.

Example for Typhoon: 
Eurofighter_24

(My process to produce these icons is just to screenshoot them in game, and resize in an image editor)

***

- If you want me to add the icons and default factions, i'll merge your PR and add them.
- However, if you want to do it, you can push the additional icons and factions files to this PR

**Comment by angelatthetomb** [comment]
Hey Khopa,

I'll get some icons and a faction added to the pull request here in the next day or two. Thanks!

**Comment by Hornet2041** [comment]
Um, your commits also say they add the VSN Prowler and F/A-18F.

**Comment by Hornet2041** [comment]
> 

image
Downloaded your latest build and got that error on trying to launch a new game.

File: game/db.py [hunk 1]
```diff
@@ -43,18 +43,28 @@
...
 from pydcs_extensions.hercules.hercules import Hercules
 from pydcs_extensions.jas39.jas39 import JAS39Gripen, JAS39Gripen_AG
+from pydcs_extensions.rafale.rafale import Rafale_B, Rafale_C
 from pydcs_extensions.su57.su57 import Su_57
 
 plane_map["A-4E-C"] = A_4E_C
+plane_map["EA-6B"] = EA_6B
+plane_map["Eurofighter"] = Eurofighter
+plane_map["FA18F_hornet"] = VSN_FA18F
+plane_map["FA18F_hornetAG"] = VSN_FA18F_AG
```
Comment (RndName) [inline game/db.py]: Neither EA-6B nor VSN_FA18F / AG are mentioned in this PR.

File: game/factions/faction.py [hunk 2]
```diff
@@ -275,6 +277,9 @@ def apply_mod_settings(self, mod_settings: ModSettings) -> Faction:
         if not mod_settings.jas39_gripen:
             self.remove_aircraft("JAS39Gripen")
             self.remove_aircraft("JAS39Gripen_AG")
+        if not mod_settings.rafale:
+            self.remove_aircraft("Rafale_B")
+            self.remove_aircraft("Rafale_C")
         if not mod_settings.su57_felon:
```
Comment (RndName) [inline game/factions/faction.py:283]: Forgot to handle the EA-6B and FA-18F.

**Review by RndName — CHANGES_REQUESTED** [review]
As already mentionend by @Hornet2041 there are unrelated mods coming in as well which are not mentionend in this PR. Please clean this one up.
Also it's kinda broken as the EA-6B and FA-18F will not be removed from the plane map when the mod is not checked.

File: resources/factions/bluefor_modern_2015.json [hunk 3]
```diff
@@ -0,0 +1,142 @@
+{
+  "country": "Combined Joint Task Forces Blue",
+  "name": "Bluefor Modern 2015",
```
Comment (RndName) [inline resources/factions/bluefor_modern_2015.json]: Why a new faction? And this one is formatted quite wrong. A lot of indents are not right.

**Comment by angelatthetomb** [comment]
Hey guys, sorry about the EA6B and FA18 references, I committed to the wrong fork. Those are in my next project.

For now, I've got this PR back to the correct commit, and from there I added icons for the Rafale B and C and the Eurofighter. I also added them to the "Bluefor Modern" faction. Forgive me, I'm just getting to know Github, never used it before. Thanks!

File: game/db.py [hunk 4]
```diff
@@ -43,18 +43,23 @@
...
+from pydcs_extensions.eurofighter.eurofighter import Eurofighter
 from pydcs_extensions.f104.f104 import VSN_F104G, VSN_F104S, VSN_F104S_AG
 from pydcs_extensions.f22a.f22a import F_22A
 from pydcs_extensions.hercules.hercules import Hercules
 from pydcs_extensions.jas39.jas39 import JAS39Gripen, JAS39Gripen_AG
+from pydcs_extensions.rafale.rafale import Rafale_B, Rafale_C
 from pydcs_extensions.su57.su57 import Su_57
 
 plane_map["A-4E-C"] = A_4E_C
+plane_map["Eurofighter"] = Eurofighter
```
Comment (DanAlbert) [inline game/db.py:55]: This ugly system was just redone. Instead, decorate mod classes with `@planemod` and be sure to add an `import` in `pydcs_extensions.__init__`. Have a look at other mods for examples.

**Comment by RndName** [comment]
PR got stale, any further changes planned or shall we close this one?

**Comment by DanAlbert** [comment]
We're closing stale PRs. If this is still needed, please update and reopen.

**Pull request closed without merging**